### DISCUSSION TITLE
Why does VScode get stack while connecting to the kernel?

### DISCUSSION THREAD

**Discussion by lmn-glzd** [microsoft/vscode-jupyter#11189] [Questions and Answers]
Hello. I am new to Python programming and I am trying to run a python code on VScode using Jupyter notebook extension. However, I am getting "Connecting to kernel: Python 2.7.16 64-bit: Activating Python Environment 'Python 2.7.16 64-bit'" message but never actually connects to kernel to run the code. However, if I uninstall and reistall the jupyter extension each time I launch VScode, it connects to kernel successfully. How can I make sure that VScode connects to the kernels without reinstalling Jupyter notebook in each launch? Can anyone please help me?

**Comment by DonJayamanne** [comment]
> nel: Python 2.7.16 64-bit: Activating Python Environment 'Python 2.7.16 64-bit'"

Unfortunately, Python 2.7 is no longer supported and its possible you are running into an issue due to the version of Pytohn used.
Do you run into this same issue with other versions of Python?

**Reply by lmn-glzd** [reply]
Yes, unfortunately, I get the same message for Python 3.9.6 version as well. It is stuck on connecting to kernel with the message "Connecting to kernel: Python 3.9.6 64-bit: Activating Python Environment 'Python 3.9.6 64-bit'"

**Reply by DonJayamanne** [reply]
> r Python 3.9.6 version as well. It is stuck on connecting to kernel with the message "Connecting to kernel: Python 3.9.6 64-bit: Activating Python Environment 'Python 3.9.6 64-bit'"

Thanks, thats  a bug.
Please could you  file a bug with the details requested in the issue template (i'm interested in the logs)

**Reply by lmn-glzd** [reply]
Hello. I think this is what you are asking for:
[Info  - 11:08:35 AM] (805) Pylance language server 2022.8.40 (pyright f444ffa5) starting
[Info  - 11:08:35 AM] (805) Server root directory: /Users/leman/.vscode/extensions/ms-python.vscode-pylance-2022.8.40/dist
[Info  - 11:08:35 AM] (805) Notebook support: LSP
[Info  - 11:08:35 AM] (805) Interactive window support: LSP
[Info  - 11:08:35 AM] (805) Starting service instance "/Users/leman/Desktop/PY0101EN-2-2-Lists.ipynb"
[Info  - 11:08:35 AM] (805) No configuration file found.
[Info  - 11:08:35 AM] (805) No pyproject.toml file found.
[Info  - 11:08:35 AM] (805) Setting pythonPath for service "/Users/leman/Desktop/PY0101EN-2-2-Lists.ipynb": "/usr/local/bin/python3"
[Warn  - 11:08:35 AM] (805) stubPath /Users/leman/Desktop/typings is not a valid directory.
[Info  - 11:08:35 AM] (805) Assuming Python version 3.9
[Info  - 11:08:35 AM] (805) Assuming Python platform Darwin
[Info  - 11:08:35 AM] (805) Searching for source files
[Info  - 11:08:36 AM] (805) Found 56 source files
[Info  - 11:08:36 AM] (805) Background analysis(1) root directory: /Users/leman/.vscode/extensions/ms-python.vscode-pylance-2022.8.40/dist
[Info  - 11:08:36 AM] (805) Background analysis(1) started

**Reply by pjsjongsung** [reply]
Hi! I also have the same issue and was wondering if my logs would also help. I am running Python 3.8.12.

Visual Studio Code (1.70.2, undefined, desktop)
Jupyter Extension Version: 2022.7.1102252217.
Python Extension Version: 2022.12.1.
No workspace folder opened.
info 11:6:7.999: ZMQ install verified.
사용자가 실험 그룹 'jupyterTest'에 속함
사용자가 실험 그룹 'jupyterEnhancedDataViewer'에 속함
info 11:6:8.483: LSP Notebooks experiment is enabled
info 11:6:51.952: Execute Cell 0 c:\Games\opto_playground.ipynb
info 11:6:51.956: Starting Jupyter Session id = 'startUsingPythonInterpreter:.jvsc74a57bd09d6eaf8865c32584f79857e931e60f77576f32be75d3d4cff32029e665ab24e8.c:\Games\miniconda3\envs\vessel_seg\python.exe.c:\Games\miniconda3\envs\vessel_seg\python.exe.-m#ipykernel_launcher' (Python Path: c:\Games\miniconda3\envs\vessel_seg, EnvType: Conda, EnvName: 'vessel_seg', Version: 3.8.12) for 'c:\Games\opto_playground.ipynb' (disableUI=false)
info 11:6:51.958: Creating raw notebook for resource 'c:\Games\opto_playground.ipynb'
info 11:6:51.958: Computing working directory for resource 'c:\Games\opto_playground.ipynb'
info 11:6:52.29: Starting raw kernel 'vessel_seg (Python 3.8.12)' for interpreter c:\Games\miniconda3\envs\vessel_seg\python.exe
info 11:6:52.108: Kernel launching with ports 9000,9001,9002,9003,9004. Start port is 9000
info 11:7:0.32: Execute Cell 1 c:\Games\opto_playground.ipynb
info 11:7:17.982: Disposing kernel .jvsc74a57bd09d6eaf8865c32584f79857e931e60f77576f32be75d3d4cff32029e665ab24e8.c:\Games\miniconda3\envs\vessel_seg\python.exe.c:\Games\miniconda3\envs\vessel_seg\python.exe.-m#ipykernel_launcher for notebook c:\Games\opto_playground.ipynb due to selection of another kernel or closing of the notebook
info 11:7:17.982: Dispose Kernel 'c:\Games\opto_playground.ipynb' associated with 'c:\Games\opto_playground.ipynb'
info 11:7:17.985: Dispose Kernel 'c:\Games\opto_playground.ipynb' associated with 'c:\Games\opto_playground.ipynb'
warn 11:7:18.2: Failed to dispose old kernel [Canceled: Canceled
	at Object.e.throwIfCanceled (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:2629322)
	at t.Kernel.createJupyterSession (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:2222056)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)]
info 11:7:18.4: Dispose Kernel 'c:\Games\opto_playground.ipynb' associated with 'c:\Games\opto_playground.ipynb'
error 11:7:18.8: VSCodeNotebookController::handleExecution, Class name = j, completed in 17977ms, has a falsy return value [Error: notebook controller is NOT associated to notebook: file:///c%3A/Games/opto_playground.ipynb
	at Object.createNotebookCellExecution (c:\Users\Jong Sung Park\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:87:104769)
	at Function.create (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:1965222)
	at Function.getOrCreate (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:1965093)
	at t.endCellAndDisplayErrorsInCell (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:1999490)
	at j.executeCell (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:2483737)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async Promise.all (index 0)
	at async j.handleExecution (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:2479129)]
info 11:7:29.179: Execute Cell 0 c:\Games\opto_playground.ipynb
info 11:7:29.179: Starting Jupyter Session id = 'startUsingPythonInterpreter:.jvsc74a57bd09d6eaf8865c32584f79857e931e60f77576f32be75d3d4cff32029e665ab24e8.c:\Games\miniconda3\envs\vessel_seg\python.exe.c:\Games\miniconda3\envs\vessel_seg\python.exe.-m#ipykernel_launcher' (Python Path: c:\Games\miniconda3\envs\vessel_seg, EnvType: Conda, EnvName: 'vessel_seg', Version: 3.8.12) for 'c:\Games\opto_playground.ipynb' (disableUI=false)
info 11:7:29.180: Creating raw notebook for resource 'c:\Games\opto_playground.ipynb'
info 11:7:29.180: Computing working directory for resource 'c:\Games\opto_playground.ipynb'
info 11:7:29.182: Starting raw kernel 'vessel_seg (Python 3.8.12)' for interpreter c:\Games\miniconda3\envs\vessel_seg\python.exe
info 11:7:29.350: Kernel launching with ports 9008,9009,9010,9011,9012. Start port is 9000
info 11:7:30.293: Execute Cell 1 c:\Games\opto_playground.ipynb
info 11:10:59.337: Disposing kernel .jvsc74a57bd09d6eaf8865c32584f79857e931e60f77576f32be75d3d4cff32029e665ab24e8.c:\Games\miniconda3\envs\vessel_seg\python.exe.c:\Games\miniconda3\envs\vessel_seg\python.exe.-m#ipykernel_launcher for notebook c:\Games\opto_playground.ipynb due to selection of another kernel or closing of the notebook
info 11:10:59.337: Dispose Kernel 'c:\Games\opto_playground.ipynb' associated with 'c:\Games\opto_playground.ipynb'
info 11:10:59.339: Dispose Kernel 'c:\Games\opto_playground.ipynb' associated with 'c:\Games\opto_playground.ipynb'
warn 11:10:59.350: Failed to dispose old kernel [Canceled: Canceled
	at Object.e.throwIfCanceled (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:2629322)
	at t.Kernel.createJupyterSession (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:2222056)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)]
info 11:10:59.352: Dispose Kernel 'c:\Games\opto_playground.ipynb' associated with 'c:\Games\opto_playground.ipynb'
error 11:10:59.355: VSCodeNotebookController::handleExecution, Class name = j, completed in 209063ms, has a falsy return value [Error: notebook controller is NOT associated to notebook: file:///c%3A/Games/opto_playground.ipynb
	at Object.createNotebookCellExecution (c:\Users\Jong Sung Park\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:87:104769)
	at Function.create (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:1965222)
	at Function.getOrCreate (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:1965093)
	at t.endCellAndDisplayErrorsInCell (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:1999490)
	at j.executeCell (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:2483737)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async Promise.all (index 0)
	at async j.handleExecution (c:\Users\Jong Sung Park\.vscode\extensions\ms-toolsai.jupyter-2022.7.1102252217\out\extension.node.js:2:2479129)]
info 11:11:7.758: Execute Cell 0 c:\Games\opto_playground.ipynb
info 11:11:7.758: Starting Jupyter Session id = 'startUsingPythonInterpreter:.jvsc74a57bd09d6eaf8865c32584f79857e931e60f77576f32be75d3d4cff32029e665ab24e8.c:\Games\miniconda3\envs\vessel_seg\python.exe.c:\Games\miniconda3\envs\vessel_seg\python.exe.-m#ipykernel_launcher' (Python Path: c:\Games\miniconda3\envs\vessel_seg, EnvType: Conda, EnvName: 'vessel_seg', Version: 3.8.12) for 'c:\Games\opto_playground.ipynb' (disableUI=false)
info 11:11:7.759: Creating raw notebook for resource 'c:\Games\opto_playground.ipynb'
info 11:11:7.759: Computing working directory for resource 'c:\Games\opto_playground.ipynb'
info 11:11:7.760: Starting raw kernel 'vessel_seg (Python 3.8.12)' for interpreter c:\Games\miniconda3\envs\vessel_seg\python.exe
info 11:11:7.931: Kernel launching with ports 9016,9017,9018,9019,9020. Start port is 9000
info 11:12:12.98: Execute Cell 1 c:\Games\opto_playground.ipynb
info 11:16:31.979: Loading webview. View is notset
info 11:16:31.980: Loading web view...
info 11:16:31.981: Webview panel created.
info 11:16:32.371: Web view react rendered

**Reply by DonJayamanne** [reply]
@pjsjongsung part could you file a bug with the details

**Comment by DonJayamanne** [comment]
anyone running into issues with connection to a kernel, i kindly request you create an issue with the necessary details
its a bug, hence bringing it up here won't help, as its not a discussion item, but a bug .
also ill need some information from the logs and the like.

once again sorry if you're running into such issues
